service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -515,6 +517,10 @@ private void checkMessageLimit(
                 .expect(ResponsePredicate.SC_OK)
                 .as(BodyCodec.jsonObject());
 
+        if (config.getQueryTimeout() > 0) {
+           request.addQueryParam("timeout", String.format("%dms", config.getQueryTimeout()));
```

Comment:
My understanding is that the query will run on Prometheus for hte given amount of time. However, since the query will be started after the request has been sent, I assume that we can run into situations where the WebClient stops waiting for the response while the query is still running. I therefore think that the timeout set on the WebClient should serve as a backstop only, i.e. it should wait for some additional time in order to allow for the query to finish in any case. WDYT?

Reply:
You are totally right, such edge case will definitely happen. Here I see several options : 
-  add _some additional time_ (for example can be 20 or 50 percentage more than _web client_ timeout ) to `timeout` query parameter ;  In my understating this will reduce probability of such edge case but still this will happen, because  this _additional time_  will certainly depend on network and maybe some Prometheus configuration.
- do not set `timeout` query parameter; Prometheus already have a global query timeout
- have additional configuration section like query `HONO_RESOURCELIMITS_PROMETHEUSBASED_QUERYPARAMETERS_TIMEOUT`, that will give for user directly access and configuration for this value.
- keep `timeout` query parameter with the value as for web client and put additional query parameter `time `  that indicate when the execution have been started (at least this is my understating regarding `time `parameter );  for more info see https://gitlab.cncf.ci/prometheus/prometheus/blob/3bce14db85998b48dde4066d9490b4f3eb4223e1/docs/querying/api.md
@sophokles73, WDYT?

Reply:
IMHO we should rely on Prometheus' global query timeout and use the vert.x WebClient's *timeout* property as a backstop. The backstop should be configurable using e.g. the `HONO_RESOURCELIMITS_PROMETHEUSBASED_QUERYTIMEOUT` variable.

Reply:
OK. Fixed , followed your suggestions.